modules/gdscript/gdscript_compiler.cpp

```diff
@@ -680,16 +680,10 @@ GDScriptCodeGenerator::Address GDScriptCompiler::_parse_expression(CodeGen &code
 				name = subscript->attribute->name;
 				named = true;
 			} else {
-				if (subscript->index->type == GDScriptParser::Node::LITERAL && static_cast<const GDScriptParser::LiteralNode *>(subscript->index)->value.get_type() == Variant::STRING) {
```

Comment:
I would actually do something like this:

```
if (subscript->index->is_constant && subscript->index->reduced_value.get_type() == Variant::STRING_NAME)
```

So it works properly for every case (using `STRING_NAME` as an index is actually the same as get named).

There's another issue in the analyzer though, which is already doing this by adding a new node to the tree (which is something I tend to avoid but it seems to have slipped):

https://github.com/godotengine/godot/blob/0ee744ff5f3cf045a2699aa4111de3200b9d70be/modules/gdscript/gdscript_analyzer.cpp#L2631-L2639

This code could be removed in favor of making the check on the compiler.